src/middlewared/middlewared/plugins/alert.py

```diff
@@ -793,6 +793,49 @@ class Config:
         """
         return await self.middleware.call("datastore.delete", self._config.datastore, id)
 
+    @accepts(
+        Int('id'),
+        Dict(
+            'alert_data',
+            Str('title', required=True),
+            Str('category', required=True, enum=list(AlertCategory.__members__)),
+            Str('level', required=True, enum=list(AlertLevel.__members__)),
+            Str('text'),
+            Str('node', default='A'),
+        )
+    )
+    async def send_alert(self, id, alert_data):
+        """
+        Send a manual alert using Alert Service `id`.
```

Comment:
Does that mean we can send alerts for unconfigured alert classes? Do we really want that?

Reply:
It was me who offered this solution. I think it's OK because user explicitly consents to receive this alert when he starts SMART test. On the other hand, if we list "SMART test finished successfully" as an alert class, user sets it's level to INFO (as it should be) and alert service level to WARNING, he will never receive this alert, which is undesirable.

Reply:
On the other hand you have to select an alert service while scheduling a manual test which kind of goes against how we do everything else, so why is this different?

While your example is undesirable it works as designed. Would probably be less confusing if we didnt have to choose which service to send to.

Reply:
> On the other hand you have to select an alert service while scheduling a manual test which kind of goes against how we do everything else, so why is this different?

If we make this UI widget collapsed dropdown with all options checked by default, it won't take much space and won't distract, nevertheless providing option to disable specific alert channels if user does not want to send "SMART test is done" alert there.

(This implies that we should change `def send_alert(self, id, alert_data):` to `def send_alert(self, ids, alert_data):` to provide ability to send to multiple sources)

99% of the users won't change any alert settings, leaving default service to "send e-mail to root when warning". With this settings they will just never be able to use this new feature.

Force button is essentially the same, just with less flexibility.

Reply:
That doesnt answer the question why this is so special.

We are going to do the same for every other thing now too? Run Cloud Sync, Run Replication, Run Cronjob, Run Rsync?

Lets discuss on Ryver. I will ping you both when I have time.